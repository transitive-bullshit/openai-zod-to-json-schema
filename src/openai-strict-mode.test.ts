import { describe, expect, test } from 'vitest'
import { z } from 'zod'

import { zodToJsonSchema } from '.'

describe('openaiStrictMode', () => {
  test('Optional properties should be set as required', () => {
    expect(
      zodToJsonSchema(
        z.object({
          foo: z.string(),
          bar: z.number().optional(),
          baz: z
            .object({
              nala: z.string().optional()
            })
            .optional()
        }),
        { openaiStrictMode: true }
      )
    ).toMatchSnapshot()
  })
})
