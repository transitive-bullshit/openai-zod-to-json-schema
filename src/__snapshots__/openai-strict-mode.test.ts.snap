// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`openaiStrictMode > Optional properties should be set as required 1`] = `
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "additionalProperties": false,
  "properties": {
    "bar": {
      "type": "number",
    },
    "baz": {
      "additionalProperties": false,
      "properties": {
        "nala": {
          "type": "string",
        },
      },
      "required": [
        "nala",
      ],
      "type": "object",
    },
    "foo": {
      "type": "string",
    },
  },
  "required": [
    "foo",
    "bar",
    "baz",
  ],
  "type": "object",
}
`;

exports[`openaiStrictMode > TODO 1`] = `
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "additionalProperties": false,
  "properties": {
    "bar": {
      "type": "number",
    },
    "baz": {
      "additionalProperties": false,
      "properties": {
        "nala": {
          "type": "string",
        },
      },
      "required": [
        "nala",
      ],
      "type": "object",
    },
    "foo": {
      "type": "string",
    },
  },
  "required": [
    "foo",
    "bar",
    "baz",
  ],
  "type": "object",
}
`;
